mslib/msui/multiple_flightpath_dockwidget.py

```diff
@@ -298,15 +349,34 @@ def save_waypoint_model_data(self, wp_model, listWidget):
             self.create_list_item(wp_model)
         self.dict_flighttrack[wp_model]["wp_data"] = wp_data
 
+    def normalize_rgb(self, rgb):
+        return tuple(channel / 255 for channel in rgb)
+
+    def get_random_color(self):
+        """
+        Get a random color from custom colors ensuring no repeats.
+        """
+        available_colors = [color for color in self.custom_colors if color not in self.used_colors]
+        if not available_colors:
+            # Reset the used colors if all colors have been used
+            self.used_colors = []
+            available_colors = self.custom_colors.copy()
+
+        selected_color = random.choice(available_colors)
+        self.used_colors.append(selected_color)
```

Comment:
This (and the reset further above) modify a class variable. This means that appending something here will affect all other MultipleFlightPathControlWidget instances. This is what I mean, for reference:
```
>>> class A:
...     x = []
... 
>>> a1 = A()
>>> a2 = A()
>>> a1.x
[]
>>> a2.x
[]
>>> a1.x.append("test123")
>>> a1.x
['test123']
>>> a2.x
['test123']
```

Is this intentional or a bug?

Reply:
yes you are right it should be an instance variable